src/main/java/edu/hm/hafner/analysis/IssueDifference.java

```diff
@@ -57,12 +70,31 @@ private void matchIssuesByFingerprint(final Report currentIssues) {
         }
     }
 
+    private <K> void addIssueToMap(final Map<K, List<Issue>> map, final K key, final Issue issue) {
+        List<Issue> issues = map.get(key);
```

Comment:
Use `putIfAbsent`

Reply:
This would put more pressure on GC creating possibly unneeded object.

Reply:
See #555